src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -928,7 +967,25 @@ internal static string ConstructInstantiation(IRuntimeMethodInfo method, TypeNam
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern RuntimeType GetDeclaringType(RuntimeMethodHandleInternal method);
+        private static extern unsafe MethodTable* GetMethodTable(RuntimeMethodHandleInternal method);
+
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "RuntimeMethodHandle_GetHandleForArray")]
+        private static unsafe partial IntPtr GetHandleForArray(MethodTable* pMT);
+
+        internal static unsafe RuntimeType GetDeclaringType(RuntimeMethodHandleInternal method)
+        {
+            Debug.Assert(!method.IsNullHandle());
+            MethodTable* pMT = GetMethodTable(method);
+            if (pMT->IsArray)
+            {
+                IntPtr handle = GetHandleForArray(pMT);
+                return RuntimeTypeHandle.GetRuntimeTypeFromHandle(handle);
+            }
+            else
+            {
```

Comment:
```suggestion
```
I do not think that this is needed. We got rid of the sharing of methodtable for arrays.